src/net.cpp

```diff
@@ -2571,6 +2583,30 @@ bool CConnman::DisconnectNode(NodeId id)
     return false;
 }
 
+bool CConnman::SetWhitelisted(NodeId id, bool fWhitelisted)
+{
+    LOCK(cs_vNodes);
+    for(CNode* pnode : vNodes) {
+        if (id == pnode->id) {
+            pnode->fWhitelisted = fWhitelisted;
+            return true;
+        }
+    }
+    return false;
+}
+
+bool CConnman::SetManualConnection(NodeId id, bool manual)
+{
+    LOCK(cs_vNodes);
+    for(CNode* pnode : vNodes) {
```

Comment:
The loop will get repeated all over the place for each field. (Already two times)

I think a better approach would be to lock at `updatepeer` level as well as fetch once for all the right CNode. And change fields directly from there.